### DISCUSSION TITLE
Footnotes?

### DISCUSSION THREAD

**Discussion by svendehens** [awcodes/filament-tiptap-editor#330] [Q&A]
Hi, I'm looking to add some footnotes functionality to the editor. Does anyone have any advice on how to go about this? 

I've been looking to either rewrite/extend the LinkAction to have a type of Footnote and dealing with the layout when rendering the content on the page. Would this be a good way to approach this?

I also saw the option to create custom Blocks. However, I'm not sure what the use cases for these Blocks exactly are? Does someone have some examples of what they've used the Blocks for? 

Thanks!

**Comment by awcodes** [comment]
What exactly are you trying to do with the footnotes? I'm not fully understanding your ask.

**Reply by svendehens** [reply]
Mainly for the user to to insert a footnote in the TipTap Editor. Since footnotes are are usually just links I thought to extend the Link Action to add this. Let's say the user has a sentence and at the end they add a footnote [1]. They would create a link on that number and go to the Link modal and select a link type of Footnote for example. Then they would just have to add the footnote text. 

The Linkaction would take care of setting the right href id automatically. Next to that it would also need to store another json object with the footnote text and the same id reference. Do you think this would be possible, from your experience with the editor?

**Reply by awcodes** [reply]
I'm not sure why you would need it other some sort of automation.

You could just make the [1] a link with an href of '#footnote-one' then later in the page you could just type the foot note and make it a link also, but set the id for it instead of an href and it will just work. Probably not optimal though.

either way though, to do what you are wanting you would need a custom extension and a custom tool other than the link modal.
